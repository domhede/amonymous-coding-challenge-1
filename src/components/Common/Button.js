import React from 'react';
import styled from 'styled-components';
import { effects } from 'helpers/style-utils';

const Button = styled.div`
  border-radius: 4px;
  padding: 16px 24px;
  color: #fff;
  background-color: #6157ff;
  color: white;
  font-size: 15px;
  text-align: center;
  font-weight: bold;
  padding: 10px 35px;
  margin: auto;
  ${props => props.withShadow ? effects.boxShadow : ''}
  &:hover {
    cursor: pointer;
    background-color: #524ad9;
  }
`;

export default (props) => <Button {...props} />;
