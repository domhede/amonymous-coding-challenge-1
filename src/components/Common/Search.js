import React, { Component } from 'react';
import styled from 'styled-components';
import { effects } from 'helpers/style-utils';

import Button from './Button';

const Wrapper = styled.div`
  display: flex;
  width: 100%;
  ${effects.boxShadow}
  padding: 0px 32px 0px;
`;
const Input = styled.input`
  margin: 32px 16px 32px 0px;
  padding: 8px;
  width: 80%;
  border-radius: 4px;
  font-size: 14px;
`;
Input.displayName = 'Input';
const Clear = styled.div`
  height: 20px;
  width: 20px;
  padding: 10px;
  border-radius: 50%;
  align-self: center;
  margin-right: 16px;
  color: #fff;
  background: #DDD;
  &:hover{
    cursor: pointer;
  }
`;
Clear.displayName = 'Clear';

const Cross = styled.path`
  stroke: black;
  fill: transparent;
  stroke-linecap: round;
  stroke-width: 5;
`;
class Search extends Component {
  constructor(props) {
    super(props);
    this.state = {
      searchTerm: ''
    }
    this.handleChange = this.handleChange.bind(this);
    this.clear = this.clear.bind(this);
  }
  handleChange(event){
    this.setState({
      searchTerm: event.target.value
    })
  }
  clear() {
    this.setState({
      searchTerm: ''
    }, () => this.props.clearFilter())
  }
  render() {
    const { filterImages } = this.props;
    const { searchTerm } = this.state;

    return (
      <Wrapper>
        <Input placeholder='Search by title' onChange={this.handleChange} value={searchTerm} />
        {searchTerm !== '' &&
          <Clear onClick={() => this.clear()}>
            <svg>
              <Cross d="M 5,5 L 15,15 M 15,5 L 5,15"></Cross>
            </svg>
          </Clear>
        }
        <Button onClick={() => filterImages(searchTerm)}>Filter</Button>
      </Wrapper>
    )
  }
}
export default Search;

