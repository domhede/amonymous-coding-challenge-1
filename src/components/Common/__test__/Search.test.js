import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Search from '../Search';
import Button from '../Button';

const mockFilter = jest.fn();
const mockClear = jest.fn();

describe('Search component', () => {
  it('renders without crashing', () => {
    shallow(
      <Search filterImages={mockFilter} clearFilter={mockClear} />
    );
  });
  it('renders correctly', () => {
    const rendered = renderer.create(
      <Search filterImages={mockFilter} clearFilter={mockClear} />
    );
    expect(rendered.toJSON()).toMatchSnapshot();
  });
  it('Should call the filterImages function when Filter Button is clicked', () => {
    const wrapper = shallow(
      <Search filterImages={mockFilter} clearFilter={mockClear} />
    );
    expect(mockFilter.mock.calls.length).toEqual(0);
    wrapper.find(Button).simulate('click');
    expect(mockFilter.mock.calls.length).toEqual(1);
  });
  it('Should call the clearFilter function when Clear Button is clicked', () => {
    const wrapper = shallow(
      <Search filterImages={mockFilter} clearFilter={mockClear} />
    );
    const input = wrapper.find('Input');
    expect(mockClear.mock.calls.length).toEqual(0);
    input.simulate('change', { target: { value: 'Hello' } })
    wrapper.find('Clear').simulate('click');
    expect(mockClear.mock.calls.length).toEqual(1);
  });
});


