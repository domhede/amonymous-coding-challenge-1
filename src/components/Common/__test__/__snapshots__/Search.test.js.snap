// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component renders correctly 1`] = `
<div
  className="sc-bwzfXH jBeist"
>
  <input
    className="sc-htpNat gsizbl"
    onChange={[Function]}
    placeholder="Search by title"
    value=""
  />
  <div
    className="sc-bdVaJa iVrSmt"
    onClick={[Function]}
  >
    Filter
  </div>
</div>
`;
